---
output: 
    html_document:
        code_folding: hide
        df_print: tibble
        toc: true
        toc_depth: 5
        toc_float: true
        css: /camp/stp/babs/working/boeings/Stefan/protocol_files/github/boeings/templates/style/style.css

always_allow_html: yes

---


```{r setup, include=FALSE}

rm(list = ls())
knitr::opts_chunk$set(
    #fig.path = "path/to/figures/",
    tidy = FALSE,
    tidy.opts = list(width.cutoff = 120),
    message = FALSE,
    warning = FALSE
)

```

```{r hpc_notes, include=FALSE}

## Get interactive session ##
#  srun --time=08:00:00 --mem=40G -p int --pty bash

# module purge;source /camp/stp/babs/working/software/modulepath_new_software_tree_2018-08-13;module load pandoc/2.2.3.2-foss-2016b;ml R/3.6.0-foss-2016b-BABS;R;

# sbatch --time=12:00:00 --wrap "module purge;source /camp/stp/babs/working/software/modulepath_new_software_tree_2018-08-13;module load pandoc/2.2.3.2-foss-2016b;ml R/3.6.0-foss-2016b-BABS;Rscript runA.r" --job-name="rA"  --mem=100G -o rA.slurm >> commands.txt

# sbatch --time=12:00:00 --wrap "module purge;source /camp/stp/babs/working/software/modulepath_new_software_tree_2018-08-13;module load pandoc/2.2.3.2-foss-2016b;ml R/3.6.0-foss-2016b-BABS;Rscript runA.r" --job-name="rA" -p hmem --mem=300G -o rA.slurm >> commands.txt

# --mem-per-cpu=14G -p hmem --pty bash

```

## Data Preparation
### Create Environment and set parameters
```{r init, eval=T, echo=T}
###############################################################################
## Create biologic Object for visualization                                  ##
singleCellClusterParameter <- 0.4
NtopGenes <- 2000
singleCellSeuratMtCutoff <- c(10)
singleCellSeuratNpcs4PCA <- 30
SeuratNrnaMaxFeatures <- 10000
SeuratNrnaMinFeatures <- 1000

## Done                                                                      ##
###############################################################################

```


```{r dbpwd, eval=T, echo=F}
#Create the environment and load a suitable version of R, e.g. so:
#library(keyring)

```

### Set Parameters
```{r set_directories, eval=T}
## Setup plot collection object
VersionPdfExt <- paste0(".V", gsub("-", "", Sys.Date()), ".pdf")

if (dir.exists("/Volumes/babs/working/boeings/")){
    hpc.mount <- "/Volumes/babs/working/boeings/"
} else if (dir.exists("Y:/working/boeings/")){
    hpc.mount <- "Y:/working/boeings/"
} else if (dir.exists("/camp/stp/babs/working/boeings/")){
    hpc.mount <- "/camp/stp/babs/working/boeings/"
} else {
    hpc.mount <- ""
}



source("assets/scTools.r")
source("assets/SBwebtools.pckg.r")



if (length(.libPaths()) > 2){
    .libPaths(.libPaths()[2:3])
}

library(dplyr)
library(ggplot2)
library(tidyverse)
library(tidyr)
library(knitr)
library(Seurat)
library(DT)


###############################################################################
## Create sampleDetail List                                                  ##

# type must be in c("TenX", "matrixFiles", "loomFiles", "hdf5Files")

sampleDetailList <- list(

    "mliGlia" = list(
        "type" = "TenX",
        "gene.column" = 1, # default 2
        "path" = paste0(hpc.mount, "Projects/pachnisv/yuuki.obata/409_sc_RNA_seq_enteric_glia_neuron_Regev_SC1038/FASTQ_files/mliGlia/outs/filtered_feature_bc_matrix"),
        "singleCellClusterParameter" = singleCellClusterParameter,
        "singleCellSeuratMtCutoff" = singleCellSeuratMtCutoff,
        "singleCellSeuratNpcs4PCA" = singleCellSeuratNpcs4PCA,
        "SeuratNrnaMaxFeatures" = SeuratNrnaMaxFeatures,
        "SeuratNrnaMinFeatures" = SeuratNrnaMinFeatures,
        "limsId"= "SCP1038",

        "singleCellChemistry" = NULL,
	    "singleCellCellrangerVersion" = NULL, 
        "machine" = NULL
    ),

    "mliNeuron" = list(
        "type" = "TenX",
        "gene.column" = 1,
        "path" = paste0(hpc.mount,"Projects/pachnisv/yuuki.obata/409_sc_RNA_seq_enteric_glia_neuron_Regev_SC1038/FASTQ_files/mliNeuron/outs/filtered_feature_bc_matrix"),
        "singleCellClusterParameter" = singleCellClusterParameter,
        "singleCellSeuratMtCutoff" = singleCellSeuratMtCutoff,
        "singleCellSeuratNpcs4PCA" = singleCellSeuratNpcs4PCA,
        "SeuratNrnaMaxFeatures" = SeuratNrnaMaxFeatures,
        "SeuratNrnaMinFeatures" = SeuratNrnaMinFeatures,
        "limsId"= "SCP1038",

        "singleCellChemistry" = NULL,
	    "singleCellCellrangerVersion" = NULL, 
        "machine" = NULL
    )
)

##                                                                           ##
###############################################################################

###############################################################################
## dbDetailList                                                              ##
dbDetailList <- list(
    "primDataDB" = "vpl_data",
    "ref.cat.db" = "reference_categories_db_new",
     "db.user" = "boeingS",
     "host" = "10.27.241.82"
)
## End dbDetailList                                                          ##
###############################################################################

###############################################################################
## Project detail list                                                       ##
projectDetailList <- list(

    "folder" = "pachnisv/yuuki.obata/409_sc_RNA_seq_enteric_glia_neuron_Regev_SC1038/",

    "lab.categories.table" = "vp_lab_categories", # default NULL
    "sra.id.vector" = "",
    "gse.id.vector" = "",
    "asf.id" = "SCP1038",
    "experiment.type" = "sc_rna_seq",   
    "species" = "mus_musculus", 
    "release" = "release-89", 

    "project_id" = "vpl409",
    "labname" = "Regev",

    "timecourse.units" = "",
    "count.table.headline" = "lg10 Expr for all Samples",
    "count.table.sidelabel" = "lg10 Expr",
    "heamap.headline.text" = "Heatmap: Row-averaged Expr"
)
## End project detail list                                                   ##
###############################################################################

###############################################################################
## Project Parameters                                                        ##
documentationParams <- list(

    "title" = "The human and mouse enteric nervous system at single cell resolution",
    "subtitle" = "Regev Lab - Cell 2020",
    "abstract" = "The enteric nervous system (ENS) coordinates diverse functions in the intestine, but has eluded comprehensive molecular characterization due to the rarity and diversity of cells. Here, we develop two methods to profile the ENS of adult mice and humans at single cell resolution: RAISIN RNA-Seq, for profiling intact nuclei with ribosome-bound mRNA, and MIRACL-Seq, for label-free enrichment of rare cell types by droplet-based profiling. The 1,187,535 nuclei in our mouse atlas include 5,068 neurons from the ileum and colon, revealing extraordinary neuron diversity. We highlight circadian expression changes in enteric neurons, show that disease-related genes are dysregulated with aging, and identify differences between the ileum and proximal/distal colon. In human, we profile 436,202 nuclei recovering 1,445 neurons and identify conserved and species-specific transcriptional programs, and putative neuro-epithelial, neuro-stromal, and neuro-immune interactions. The human ENS expresses risk genes for neuropathic, inflammatory, and extra-intestinal diseases, suggesting neuronal contributions to disease. In this analysis the mouse mli Neuron and mli Glia subset is re-analysed."

)


## Done Project Params                                                       ##
###############################################################################


###############################################################################
## Project detail list                                                       ##
scDetailList <- list(
    "NtopGenes" =  NtopGenes,
    "singleCellClusterParameter" = singleCellClusterParameter,
    "singleCellClusterString" = paste0("integrated_snn_res.", singleCellClusterParameter),
    "scIntegrationMethod" = "SCT", # "SCT" or "standard"
    "scNintegrationFeatures" = 3000,
    "primReduction" = "umap"
)
## End project detail list                                                   ##
###############################################################################

###############################################################################
## Reference Table List                                                      ##
referenceTableList = list(
    "Hallmark Signatures" = "mysigdb_h_hallmarks",
    "Pathways" = "mysigdb_c2_1329_canonical_pathways",
    "GO-BP" = "mysigdb_c5_BP",                               
    "GO-MF" = "mysigdb_c5_MF",
    "TF Motifs" = "TRANSFAC_and_JASPAR_PWMs",
    "Protein Complexes" = "networkcategories",
    "GO-BP" = "GO_Biological_Process_2017",
    "Immunologic Signatures" = "mysigdb_c7_immunologic_signatures",
    "LINCS Down" = "LINCS_L1000_Chem_Pert_down",
    "LINCS Up" = "LINCS_L1000_Chem_Pert_up",
    "Allen Brain Atlas" = "Allen_Brain_Atlas",
    "Cell Type Signatures" = "mysigdb_sc_sig",
    "Cell Type Signatures" = "cibersort_L22"
)

##############################################################################
## Tables for signature enrichment                                          ##
clusterSigEnrichmentList = list(
    "Cell Type Signatures" = "mysigdb_sc_sig",
    "Cell Type Signatures" = "cibersort_L22",
    "GO-MF" = "mysigdb_c5_MF",
    "Pathways" = "mysigdb_c2_1329_canonical_pathways",
    "Hallmarks" = "mysigdb_h_hallmarks",
    "Allen_Brain_Atlas" = "Allen_Brain_Atlas"
)

    # mysigdb_sc_sig
    # cibersort_L22
    # Allen_Brain_Atlas                            |
    # CORUM                                        |
    # ChEA_2016                                    |
    # DEPOD_phosphatase_substrates                 |
    # ENCODE_TF_ChIP_seq_2015                      |
    # GO_Biological_Process_2017                   |
    #LINCS_L1000_Chem_Pert_down                   |
    #LINCS_L1000_Chem_Pert_down_backup            |
    # LINCS_L1000_Chem_Pert_up                     |
    # LINCS_L1000_Chem_Pert_up_backup              |
    # NCBI_homologene_table                        |
    # Old_CMAP_down                                |
    # Old_CMAP_up                                  |
    # SGP_from_GEO_up_down_combined                |
    # SILAC_Phosphoproteomics                      |
    # TRANSFAC_and_JASPAR_PWMs                     |
    # UK_Biobank_GWAS                              |
    # ag_lab_categories                            |
    # as_lab_categories                            |
    # bader_lab_hESC_reference                     |
    # bt_lab_categories                            |
    # cat_selection_default                        |
    # cs_lab_categories                            |
    # da_lab_categories                            |
    # es_lab_categories                            |
    # esl111_cat_reference_db_table                |
    # et_lab_categories                            |
    # exploration_categories                       |
    # fg_lab_categories                            |
    # fi_lab_categories                            |
    # gk_lab_categories                            |
    # innateDB_PPI                                 |
    # jb_lab_categories                            |
    # js_lab_categories                            |
    # kn_lab_categories                            |
    # mysigdb_c1_positional                        |
    # mysigdb_c2_1329_canonical_pathways           |
    # mysigdb_c2_KEGG                              |
    # mysigdb_c2_REACTOME                          |
    # mysigdb_c2_biocarta                          |
    # mysigdb_c2_chemical_and_genetic_pertubations |
    # mysigdb_c3_TF_targets                        |
    # mysigdb_c3_miRNA_targets 
# et_lab_categories                            |
# | exploration_categories                       |
# | fg_lab_categories                            |
# | fgl391_cat_reference_db_table                |
# | fi_lab_categories                            |
# | gk_lab_categories                            |
# | innateDB_PPI                                 |
# | jb_lab_categories                            |
# | js_lab_categories                            |
# | kn_lab_categories                            |
# | mysigdb_c1_positional                        |
# | mysigdb_c2_1329_canonical_pathways           |
# | mysigdb_c2_KEGG                              |
# | mysigdb_c2_REACTOME                          |
# | mysigdb_c2_biocarta                          |
# | mysigdb_c2_chemical_and_genetic_pertubations |
# | mysigdb_c3_TF_targets                        |
# | mysigdb_c3_miRNA_targets                     |
# | mysigdb_c5_BP                                |
# | mysigdb_c5_CC                                |
# | mysigdb_c5_MF                                |
# | mysigdb_c6_oncogenic_signatures              |
# | mysigdb_c7_immunologic_signatures            |
# | mysigdb_h_hallmarks                          |
# | networkcategories                            |
# | nl_lab_categories                            |
# | pa_lab_categories                            |
# | pb_lab_categories                            |
# | pfam_interpro                                |
# | pp_lab_categories                            |
# | project_db_table                             |
# | project_db_table_backup                      |
# | project_description_table                    |
# | pt_lab_categories                            |
# | re_lab_categories                            |
# | reference_categories_db_new                  |
# | rl_lab_categories                            |
# | sb_lab_categories                            |
# | sc_lab_categories                            |
# | sl_lab_categories                            |
# | sl_lab_categories_backup                     |
# | ss_lab_categories                            |
# | st_lab_categories                            |
# | temp_categories                              |
# | vp_lab_categories                            |
# | vt_lab_categories

## Done                                                                      ##
###############################################################################

# Species has to be "mus_musculus", "homo_sapiens", "danio_rerio" 
# release-86, release-89

Obio = new(
    "bioLOGIC",
    clusterSigEnrichmentList = clusterSigEnrichmentList,
    documentationParams = documentationParams,
    dbDetailList = dbDetailList,        
    projectDetailList = projectDetailList,
    sampleDetailList = sampleDetailList, 
    referenceTableList = referenceTableList,
    parameterList = list(

        "debugReduce" = NULL, # Default NULL

        "lab.categories.table" = projectDetailList$lab.categories.table, # default NULL
        "folder" = projectDetailList$folder,
        "sra.id.vector" = "",
        "gse.id.vector" = "",
        "addDmaps" = TRUE,

       "lims.id"= "",
        "asf.id" = "",

        "machine" = NULL,
        "experiment.type" = "sc_rna_seq",   
        "species" = projectDetailList$species, 
        "release" = "release-89", 
        "project_id" = projectDetailList$project_id,
        "labname" = projectDetailList$labname,
        "db.user" =  dbDetailList$db.user,
        "host" = dbDetailList$host,
        "timecourse.units" = "",
        "count.table.headline" = "lg10 Expr for all Samples",
        "count.table.sidelabel" = "lg10 Expr",
        "heamap.headline.text" = "Heatmap: Row-averaged Expr",
        "loadR" = "module purge;source /camp/stp/babs/working/software/modulepath_new_software_tree_2018-08-13;module load pandoc/2.2.3.2-foss-2016b;ml R/3.6.0-foss-2016b-BABS;R;",
        "pathToSeqStorageFolder" = NULL,
        "matrixFiles" = NULL,
        "loomFiles" = NULL,
        "addFullTPMtable" = FALSE,
        "hpcMount" = "",
        "parallelProcessing" = FALSE,
        "timeseries" = TRUE,
        "NtopGenes" =  NtopGenes,
        "singleCellClusterParameter" = singleCellClusterParameter,
        "singleCellClusterString" = paste0("integrated_snn_res.", singleCellClusterParameter),
        "singleCellRead10XgeneColumn" = 1, # default is 2
        "singleCellPercExpressedMinCutOff" = 10,
        "singleCellTranscriptome"="GRCh38",
	    "singleCellChemistry" = NULL,
	    "singleCellCellrangerVersion" = NULL, 
        "singleCellSeuratMtCutoff" = singleCellSeuratMtCutoff,
        "singleCellSeuratNpcs4PCA" = singleCellSeuratNpcs4PCA,
        "scIntegrationMethod" = "SCT", # "SCT" or "standard"
        "scNintegrationFeatures" = scDetailList$scNintegrationFeatures,
        "SeuratNrnaMaxFeatures" = SeuratNrnaMaxFeatures,
        "SeuratNrnaMinFeatures" = SeuratNrnaMinFeatures,
        "primReduction" = "umap",
        "qcReports" = list(),
        "catRefFile" = "/camp/stp/babs/working/boeings/Projects/pachnisv/song.chng/330_10xscRNAseq_DIV4_DIV11_DIV20_SC19069/basedata/vpl330.mm.AUC.gene.cats.txt",
        referenceTableList = referenceTableList
    )
)


Obio <- setMountingPoint(Obio)
Obio <- setAnalysisPaths(Obio)
Obio <- setCrickGenomeAndGeneNameTable(Obio)
Obio <- createAnalysisFolders(
    Obio,
    baseDir="/camp/stp/babs/working/boeings/Projects/",
    localBaseDir = paste0(hpc.mount, "Projects/")
)
Obio <- setDataBaseParameters(Obio)


Obio@parameterList[["reportFigDir"]] <- paste0(Obio@parameterList$localWorkDir,Obio@parameterList$project_id, "/report_figures/")

####

FN <- paste0(hpc.mount, "Projects/reference_data/documentation/BC.parameters.txt")
dbTable <- read.delim(
    FN, 
    sep = "\t",
    stringsAsFactors = F
)

db.pwd <- as.vector(dbTable[1,1])

## Create outputfolders ##
if (!dir.exists(paste0(Obio@parameterList$localWorkDir,Obio@parameterList$project_id))){
    dir.create(paste0(Obio@parameterList$localWorkDir,Obio@parameterList$project_id))
}

if (!dir.exists(Obio@parameterList$reportFigDir)){
    dir.create(Obio@parameterList$reportFigDir)
}


figureCount <- 1

## Load R module load R/3.5.1-foss-2018b ##
#setwd(Obio@parameterList$localWorkDir)
```



```{r saveobject, eval=TRUE, echo=T, results=F}
### Will save Obio object here, so it can be re-used with different parameters
save(Obio, 
     file = paste0(
         Obio@parameterList$localWorkDir,
         Obio@parameterList$project_id,
         ".bioLOGIC.Robj"
     )
)

print("R bioLOGIC single cell object initialized.")

```

## QC Section


```{r create_QC_Table, echo=TRUE, eval=TRUE, warning=FALSE, results=F}

tableCreated <- FALSE
sampleNames <- names(Obio@sampleDetailList)

for (i in 1:length(sampleNames)){
    if (Obio@sampleDetailList[[sampleNames[i]]]$type == "TenX"){
        baseFN <- Obio@sampleDetailList[[sampleNames[i]]]$path
        summaryFN <- gsub("filtered_feature_bc_matrix", "metrics_summary.csv", baseFN)
        if (file.exists(summaryFN)){
            dfTemp <- read.csv(summaryFN)
            dfTemp[["sampleName"]] <- sampleNames[i]
            if (!tableCreated){
                tableCreated = TRUE
                dfRes <- dfTemp
            } else {
                dfRes <- rbind(
                    dfRes,
                    dfTemp
                )
            }
        }
    }
}

if (exists("dfRes")){
dfRes <- data.frame(t(dfRes))
colVec <- as.vector(t(dfRes["sampleName",]))
names(dfRes) <- colVec


dfRes[["Parameter"]] <- row.names(dfRes)
dfRes <- dfRes[!(row.names(dfRes) %in% c("sampleName")),]
row.names(dfRes) <- NULL
colVec <- c("Parameter", colVec)
dfRes <- dfRes[,colVec]
dfRes$Parameter <- gsub("[.]", " ", dfRes$Parameter)


###############################################################################
## Write table to Excel File                                                 ##

createXLSXoutput(
    dfTable = dfRes,
    outPutFN = paste0(Obio@parameterList$outputDir, "/",Obio@parameterList$project_id, "_QC_Parameter_Table.xlsx"),
    tableName = paste0(Obio@parameterList$project_id, "_QC_Parameter_Table")
)

## Done creating Excel output table                                          ##
###############################################################################

Obio@parameterList$reportTableDir <- gsub("report_figures", "report_tables",Obio@parameterList$reportFigDir)

if (!dir.exists(Obio@parameterList$reportTableDir)){
    dir.create(Obio@parameterList$reportTableDir)
}

FNbase <- paste0(Obio@parameterList$project_id, "_QC_Parameter_Table.xlsx")
FN <- paste0(Obio@parameterList$reportTableDir, FNbase)
FNrel <- paste0("report_tables/", FNbase)


tabDownloadLink <- paste0("The quality measures table can be downloaded [here](",FNrel,")")

tabLegend = paste0(
    "**Table: ** QC Parameters for all 10X single-cell samples in this experiment. ",
    tabDownloadLink
)

chnkVec <- paste0(
        #"#### ", names(dtList),
        "\n```{r QC_datatable, results='asis', echo=F, eval=TRUE, warning=FALSE, fig.cap='",
        tabLegend,"'}\n",
        "\n",
        "\n datatable(dfRes,rownames = FALSE,  escape = FALSE)",
        "\n cat(  '\n')",
        "\n\n\n```\n"   
    )
    tabVar <- "### Create QC Table"
}   else {
    tabVar <- ""
}
## Done creating one table per cluster                                      ##
##############################################################################
```

`r tabVar`
```{r render_QCTable, echo=TRUE, eval=TRUE, warning=FALSE, results="asis"}

if (exists("dfRes")){
    cat(paste(knit(text = chnkVec, quiet = T), collapse = '\n'))
    
} 
```





```{r QC_raw_data, echo=TRUE, eval=TRUE, warning=FALSE, results=F}

###############################################################################
## Create Cell Ranger QC Plots                                               ##
if (exists("dfRes")){
## Add if clause to check 10X
resList <- doCRplots(
    obj = Obio,
    figureCount = figureCount,
    VersionPdfExt = VersionPdfExt,
    tocSubLevel = 4,
    dotsize = 0.5
) 

plotListQC1 <- resList$plotListQC1
chnkVec <- resList$chnkVec
figureCount <- resList$figureCount
    if (length(plotListQC1) > 3){
        
        tabVar <- "### CellRanger QC {.tabset .tabset-fade .tabset-dropdown}"
    } else {
        
        tabVar <- "### CellRanger QC {.tabset .tabset-fade .tabset-pills}"
    }

    
        
} else {
    tabVar <- ""
}

## Done create cellRanger QC plots                                           ##
###############################################################################


 
```


`r tabVar`

```{r Plot_tsne_data_plotting, echo=TRUE, eval=TRUE, warning=FALSE, results="asis"} 
if (exists("dfRes")){
cat(paste(knit(text = chnkVec, quiet = T), collapse = '\n'))
}

```


## Determining Parameters for Subsequent Single-cell Analysis
### Make Sample List for single-sample assessment
```{r start_seurat, echo=TRUE, eval=TRUE, warning=FALSE, results=F}

SampleList <- createSampleListQC(
    obj = Obio,
    reduce = Obio@parameterList$debugReduce    
)

print(paste0("Cell Recuction: ", Obio@parameterList$debugReduce))
lapply(SampleList, dim)

```


### Create PercentMT Historgrams Plots {.tabset .tabset-fade .tabset-pills}
```{r percent_mt, echo=TRUE, eval=TRUE, warning=FALSE, results=F}
###############################################################################
## Do percent mt plots                                                       ##
resList <- doPercMT_plotSL(
        SampleList = SampleList,
        obj = Obio,
        figureCount = figureCount,
        VersionPdfExt = ".pdf",
        tocSubLevel = 4
)


plotListRF <- resList$plotListRF
chnkVec <- resList$chnkVec
figureCount <- resList$figureCount

## Done create cellRanger QC plots                                           ##
###############################################################################

```

```{r Plot_percentMT_hist, echo=TRUE, eval=TRUE, warning=FALSE, results="asis"} 

cat(paste(knit(text = chnkVec, quiet = T), collapse = '\n'))

```

### Create RNA Feature Historgrams Plots {.tabset .tabset-fade .tabset-pills}
```{r nFeature_data, echo=TRUE, eval=TRUE, warning=FALSE, results=F}
###############################################################################
## Do percent mt plots                                                       ##
resList <- doRNAfeat_plotSL(
        SampleList,
        obj = Obio,
        figureCount = figureCount,
        VersionPdfExt = ".pdf",
        tocSubLevel = 4
)

plotListRF <- resList$plotListRF
chnkVec <- resList$chnkVec
figureCount <- resList$figureCount

## Done create cellRanger QC plots                                           ##
###############################################################################

```

```{r Plot_nFeatureRNA_hist, echo=TRUE, eval=TRUE, warning=FALSE, results="asis"} 

cat(paste(knit(text = chnkVec, quiet = T), collapse = '\n'))

```

### Plot UMAP Per Sample {.tabset .tabset-fade .tabset-pills}
```{r UMAP_data, echo=TRUE, eval=TRUE, warning=FALSE, results=F}
###############################################################################
## Do percent mt plots                                                       ##
resList <- doUMAP_plotSL(
        SampleList,
        obj = Obio,
        figureCount = figureCount,
        VersionPdfExt = ".pdf",
        tocSubLevel = 4,
        dotsize = 0.5
)

plotListSQCUMAP <- resList$plotListSQCUMAP
chnkVec <- resList$chnkVec
figureCount <- resList$figureCount

## Done create cellRanger QC plots                                           ##
###############################################################################

```

```{r Plot_UMAP, echo=TRUE, eval=TRUE, warning=FALSE, results="asis"} 

cat(paste(knit(text = chnkVec, quiet = T), collapse = '\n'))

```


```{r UMAP_data_CC, echo=TRUE, eval=TRUE, warning=FALSE, results=F}
###############################################################################
## Do percent mt plots                                                       ##
resList <- doUMAP_cellCyle(
        SampleList,
        obj = Obio,
        figureCount = figureCount,
        VersionPdfExt = ".pdf",
        tocSubLevel = 4,
        dotsize = 0.5,
        cellCycleRefFile = paste0(hpc.mount, "Projects/reference_data/cell_cycle_vignette_files/nestorawa_forcellcycle_expressionMatrix.txt")
)


plotList <- resList$plotList
chnkVec <- resList$chnkVec
figureCount <- resList$figureCount

## Done create cellRanger QC plots                                           ##
###############################################################################

```

### Plot Cell Cycle UMAP Per Sample {.tabset .tabset-fade .tabset-pills}
```{r Plot_UMAP_CC, echo=TRUE, eval=TRUE, warning=FALSE, results="asis"} 

cat(paste(knit(text = chnkVec, quiet = T), collapse = '\n'))

```

### Plot Per Sample Cell Cycle Estimate {.tabset .tabset-fade .tabset-pills}
```{r cell_cyle_hist_data, echo=TRUE, eval=TRUE, warning=FALSE, results=F}
###############################################################################
## Do percent mt plots                                                       ##
resList <- doCellCycleBarchart(
        SampleList,
        obj = Obio,
        figureCount = figureCount,
        VersionPdfExt = ".pdf",
        tocSubLevel = 4,
        cellCycleRefFile = paste0(hpc.mount, "Projects/reference_data/cell_cycle_vignette_files/nestorawa_forcellcycle_expressionMatrix.txt")
)

plotList <- resList$plotList
chnkVec <- resList$chnkVec
figureCount <- resList$figureCount

## Done create cellRanger QC plots                                           ##
###############################################################################

```

```{r Plot_cellCycleBarchart, echo=TRUE, eval=TRUE, warning=FALSE, results="asis"} 

cat(paste(knit(text = chnkVec, quiet = T), collapse = '\n'))

```

### Plot Per Sample Doublet Estimate {.tabset .tabset-fade .tabset-pills}
```{r DF_data, echo=TRUE, eval=TRUE, warning=FALSE, results=F}
###############################################################################
## Do percent mt plots                                                       ##
resList <-  doDF_plotSL(
        SampleList,
        obj = Obio,
        figureCount = figureCount,
        VersionPdfExt = ".pdf",
        tocSubLevel = 4,
        dotsize = 0.5
)
    
 

plotListDF <- resList$plotListDF
chnkVec <- resList$chnkVec
figureCount <- resList$figureCount
Obio@dataTableList[["DF_resultlist"]] <- resList$addList
        
## Done create cellRanger QC plots                                           ##
###############################################################################

```

```{r Plot_DF, echo=TRUE, eval=TRUE, warning=FALSE, results="asis"} 

cat(paste(knit(text = chnkVec, quiet = T), collapse = '\n'))

```

### Plot Per Sample MT Percentage {.tabset .tabset-fade .tabset-pills}
```{r UMAP_percentage, echo=TRUE, eval=TRUE, warning=FALSE, results=F}
###############################################################################
## Do percent mt plots                                                       ##

resList <- doUMAP_plot_percMT(
    SampleList,
    obj =  Obio,
    figureCount = figureCount,
    VersionPdfExt = ".pdf",
    tocSubLevel = 4,
    dotsize = 0.5
)
 

plotListUMT <- resList$plotListUMT
chnkVec <- resList$chnkVec
figureCount <- resList$figureCount

        
## Done create cellRanger QC plots                                           ##
###############################################################################

```

```{r Plot_percent_MT, echo=TRUE, eval=TRUE, warning=FALSE, results="asis"} 

cat(paste(knit(text = chnkVec, quiet = T), collapse = '\n'))

```

### Plot Per Sample nFeatures RNA {.tabset .tabset-fade .tabset-pills}
```{r UMAP_nFeatures, echo=TRUE, eval=TRUE, warning=FALSE, results=F}
###############################################################################
## Do percent mt plots                                                       ##

resList <- doUMAP_plot_nFeatRNA(
    SampleList,
    obj =  Obio,
    figureCount = figureCount,
    VersionPdfExt = ".pdf",
    tocSubLevel = 4,
    dotsize = 0.5
)
 

plotListNC <- resList$plotListNC
chnkVec <- resList$chnkVec
figureCount <- resList$figureCount

        
## Done create cellRanger QC plots                                           ##
###############################################################################

```

```{r Plot_nFeatureRNA, echo=TRUE, eval=TRUE, warning=FALSE, results="asis"} 

cat(paste(knit(text = chnkVec, quiet = T), collapse = '\n'))

```



```{r saveobject_end, eval=TRUE, echo=T, results=F}
### Will save Obio object here, so it can be re-used with different parameters
save(Obio, 
     file = paste0(
         Obio@parameterList$localWorkDir,
         Obio@parameterList$project_id,
         ".bioLOGIC.Robj"
     )
)

print("R bioLOGIC single cell object initialized.")

```

```{r createIntegrated, eval=TRUE, echo=T, results=F}
### Will save Obio object here, so it can be re-used with different parameters
###############################################################################
## Create sample list filtered on MT and norm_counts_RNA                     ##
SampleList <- createNormSampleList(
    obj = Obio,
    reduce = Obio@parameterList$debugReduce # Default is NULL
)

print(paste0("Cell Recuction: ", Obio@parameterList$debugReduce))
lapply(SampleList, dim)
## Done                                                                      ##
###############################################################################

###############################################################################
## Add doublet annotation, if present, to meta data                          ##

pos <- grep("DF_resultlist", names(Obio@dataTableList))

if (length(pos) > 0){
    sampleNames <- names(SampleList)
    for (i in 1:length(SampleList)){
        dfAdd <- Obio@dataTableList[["DF_resultlist"]][[sampleNames[i]]]
        row.names(dfAdd) <- gsub("-1", "",row.names(dfAdd))
        dfAdd <- dfAdd[row.names(dfAdd) %in% row.names(SampleList[[i]]@meta.data),]
        
        SampleList[[i]] <- addDf2seuratMetaData(
            obj = SampleList[[i]],
            dfAdd = dfAdd
        )
    }
}

## Done                                                                      ##
###############################################################################

###############################################################################
## Integrate Datasets                                                        ##
if (length(SampleList) > 1){
    if (Obio@parameterList$scIntegrationMethod == "SCT"){
        
        if (length(grep("scNintegrationFeatures", names(Obio@parameterList))) == 0){
            Obio@parameterList$scNintegrationFeatures = 3000
        }
        
        library(future)
        options(future.globals.maxSize = 14000 * 1024^2)
        plan("multiprocess", workers = 30)
        
        sample.features <- SelectIntegrationFeatures(
            object.list = SampleList, 
            nfeatures = Obio@parameterList$scNintegrationFeatures
        )
        SampleList <- PrepSCTIntegration(
            object.list = SampleList, 
            anchor.features = sample.features, 
            verbose = FALSE
        )
        
        sampleAnchors <- FindIntegrationAnchors(
            object.list = SampleList, 
            normalization.method = "SCT", 
            anchor.features = sample.features, 
            verbose = FALSE
        )
        
        OsC <- IntegrateData(
            anchorset = sampleAnchors, 
            normalization.method = "SCT", 
            verbose = FALSE
        )
        detach("package:future", unload=TRUE)
        
    } else {
    
        sampleAnchors <- FindIntegrationAnchors(
            object.list = SampleList, 
            dims = 1:30
        ) 


        OsC <- IntegrateData(
            #features.to.integrate = geneIntersectVec,
            anchorset = sampleAnchors, 
            dims = 1:30
        )
    }
    Obio@dataTableList$referenceList[["sampleAnchors"]] <- as.vector(sort(sampleAnchors@anchor.features))
} else {
    OsC <- SampleList[[1]]
}

Idents(OsC) <- factor(Idents(OsC), levels = names(Obio@sampleDetailList))
OsC@meta.data$sampleID <- factor(OsC@meta.data$sampleID, levels = names(Obio@sampleDetailList))

OsC@meta.data[["cellID"]] <- row.names(OsC@meta.data)

save(OsC,
    file = paste0(
         Obio@parameterList$localWorkDir,
         Obio@parameterList$project_id,
        ".Seurat.Robj"
     )
)

print("R bioLOGIC single cell object initialized.")

```


### Most up/down-regulated genes
### Cluster gene set enrichment

## Documentation
```{r documentation, eval=TRUE, echo=T, results=T}
sessionInfo()
```

```{r create_report_params, eval=T, results="asis"}

## Try to retrieve project data from db ##
library(RMySQL)
db.pwd2 <- "_asf_"
db.user2 <- "asf"
host2 <- "ms1.thecrick.org"
projectParams <- Obio@documentationParams

tryCatch({
    dbDB = dbConnect(drv = RMySQL::MySQL(), user = db.user2, password = db.pwd2, host = host2, dbname = "asf");
dfProposal =  dbGetQuery(dbDB, paste0("SELECT * FROM asf_proposals WHERE project_name ='",Obio@parameterList$lims.id,"'"));
dbDisconnect(dbDB)
  }, error = function(x) {
    message("Project Database could not be reached or has no entry in Obio@parameterList$lims.id for this analysis.")
   
})

if (nrow(dfProposal) == 1){
    if (!is.na(dfProposal[1,"ProjectAlias"]) & dfProposal[1,"ProjectAlias"] != ""){
        projectParams[["title"]] = paste0(dfProposal[1,"ProjectAlias"], " - ", dfProposal[1,"project_name"])
    }
    
    if (!is.na(dfProposal[1,"project_user"]) & dfProposal[1,"project_user"] != ""){
        projectParams[["subtitle"]] = paste0(dfProposal[1,"user_lab"], " Lab - ", dfProposal[1,"project_user"])
        projectParams[["subtitle"]] <- gsub("^ Lab - ", "", projectParams[["subtitle"]])
        
    }
    
    if (!is.na(dfProposal[1,"proposal_text"]) & dfProposal[1,"proposal_text"] != ""){
        projectParams[["abstract"]] = dfProposal[1,"proposal_text"]
       
        
    }
}
   
## Escape all special characters
projectParams <- lapply(
  projectParams, function(x) 
  #gsub("([.|()\\^{}+$*?]|\\[|\\])", "\\\\\1", x)
  gsub("([.|()/\\^{}+$*?]|\\[|\\])", " ", x)
) 

projectParams <- lapply(
  projectParams, function(x) 
  #gsub("([.|()\\^{}+$*?]|\\[|\\])", "\\\\\1", x)
  gsub("\\\n", " ", x)
) 


#projectParams$title <- "Title"
projectParams$abstract <- "This is the QC section."
#projectParams$subtitle <- "Abstract"

```

---
title: "`r projectParams$title`"
subtitle:  "`r projectParams$subtitle`"
author:
    - Stefan Boeing^[The Francis Crick Institute, stefan.boeing@crick.ac.uk]
date: 'Compiled: `r format(Sys.Date(), "%B %d, %Y")`'

abstract: |
    "`r projectParams$abstract`"

output: 
    html_document:
        code_folding: hide
        df_print: tibble
        toc: true
        toc_depth: 5
        toc_float: true
        css: /camp/stp/babs/working/boeings/Stefan/protocol_files/github/boeings/templates/style/style.css

always_allow_html: yes


---